### PR TITLE
Reload adjacent rows using original indices, not final indices.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/Cells/ConversationViewCell.m [hunk 1]
```diff
@@ -35,53 +35,10 @@ - (void)setIsCellVisible:(BOOL)isCellVisible
     _isCellVisible = isCellVisible;
 
     if (isCellVisible) {
-        [self forceLayoutImmediately];
+        [self layoutIfNeeded];
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/Cells/ConversationViewCell.m]: Reverting this change; I think I misdiagnosed the issue and it should be fixed by the other changes in this PR.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 2]
```diff
@@ -168,7 +168,7 @@ @interface ConversationViewController () <AVAudioPlayerDelegate,
 @property (nonatomic, readonly) ConversationViewLayout *layout;
 
 @property (nonatomic) NSArray<ConversationViewItem *> *viewItems;
-@property (nonatomic) NSMutableDictionary<NSString *, ConversationViewItem *> *viewItemMap;
+@property (nonatomic) NSMutableDictionary<NSString *, ConversationViewItem *> *viewItemCache;
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: Renamed the cache.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 3]
```diff
@@ -2920,7 +2920,7 @@ - (void)yapDatabaseModified:(NSNotification *)notification
                         rowChange.finalIndex);
                     [self.collectionView insertItemsAtIndexPaths:@[ rowChange.newIndexPath ]];
                     // We don't want to reload a row that we just inserted.
-                    [rowsThatChangedSize removeObject:@(rowChange.finalIndex)];
+                    [rowsThatChangedSize removeObject:@(rowChange.originalIndex)];
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: The key fix: Reload actions should use _original_ indices, not _final_ indices.

File: Signal/src/ViewControllers/ConversationView/ConversationViewController.m [hunk 4]
```diff
@@ -2970,59 +2970,31 @@ - (void)yapDatabaseModified:(NSNotification *)notification
 
     DDLogVerbose(@"self.viewItems.count: %zd -> %zd", oldViewItemCount, self.viewItems.count);
 
-    BOOL shouldReloadCollection = [self shouldReloadCollection:rowChanges];
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/ConversationViewController.m]: We don't need this hack either now, I believe.  Moves are working properly now that we're reloading with the original indices.

File: Signal/src/ViewControllers/ConversationView/Cells/ConversationViewCell.m (hunk 1 above)
Comment (michaelkirk) [reply Signal/src/ViewControllers/ConversationView/Cells/ConversationViewCell.m]: 🎉

**Pull request merged**